File: frontend/static/js/merchant.js
// 店家端逻辑（默认店铺 id=1, owner id=1）
const app=document.getElementById('view');
const SHOP_ID=1;
const views={dash:renderDash,schedule:renderSchedule,scripts:renderScripts,dms:renderDMs};
function go(name){document.querySelectorAll('.tabbar a').forEach(a=>a.classList.toggle('active',a.dataset.v===name));(views[name]||renderDash)();window.scrollTo(0,0);}

async function renderDash(){
  app.innerHTML=`<div class="wrap"><div class="sec-title">📊 经营看板</div><div id="dash" class="loading">加载中…</div></div>`;
  const d=await API.get(`/api/merchant/shops/${SHOP_ID}/dashboard`);
  document.getElementById('dash').innerHTML=`
    <div class="stat-grid">
      <div class="stat"><div class="num">${d.total_sessions}</div><div class="lbl">总场次</div></div>
      <div class="stat"><div class="num">${d.fill_rate}%</div><div class="lbl">满场率</div></div>
      <div class="stat"><div class="num">${d.total_players}</div><div class="lbl">累计玩家</div></div>
      <div class="stat"><div class="num">¥${d.revenue}</div><div class="lbl">预估流水</div></div>
    </div>
    <div class="sec-title">🔥 热门剧本</div>
    ${(d.hot_scripts||[]).map(h=>`<div class="card row" style="justify-content:space-between"><b>${esc(h.title)}</b><span class="gold">${h.count}场</span></div>`).join('')||'<div class="empty">暂无数据</div>'}`;
}

async function renderSchedule(){
  app.innerHTML=`<div class="wrap"><div class="sec-title">📅 排期管理 <span class="more" onclick="openNew()">＋ 开场</span></div><div id="sch" class="loading">加载中…</div></div>`;
  const list=await API.get(`/api/merchant/shops/${SHOP_ID}/schedule`);
  document.getElementById('sch').innerHTML=list.map(s=>`
    <div class="card">
      <div class="row" style="justify-content:space-between"><b>${esc(s.script_title)}</b>
        <span class="status ${s.status}">${statusText(s.status)}</span></div>
      <div class="meta">🕐 ${s.start_time} · DM ${esc(s.dm_name)} · ¥${s.price}</div>
      <div class="progress"><i style="width:${Math.round(s.joined_players/s.need_players*100)}%"></i></div>
      <div class="row" style="justify-content:space-between;align-items:center">
        <span class="muted" style="font-size:13px">${s.joined_players}/${s.need_players} 人</span>
        <div>
          <button class="btn ghost" style="padding:5px 12px;font-size:12px" onclick="setStatus(${s.id},'confirmed')">确认成团</button>
          <button class="btn ghost" style="padding:5px 12px;font-size:12px" onclick="setStatus(${s.id},'cancelled')">取消</button>
        </div>
      </div>
    </div>`).join('')||'<div class="empty">暂无排期，点右上角开场</div>';
}
function statusText(s){return {recruiting:'招募中',full:'已满',confirmed:'已成团',finished:'已结束',cancelled:'已取消'}[s]||s;}
async function setStatus(id,st){await API.post(`/api/merchant/schedule/${id}/status`,{status:st});renderSchedule();}

async function openNew(){
  const scripts=await API.get('/api/player/scripts');
  const dms=await API.get(`/api/merchant/shops/${SHOP_ID}/dms`);
  app.innerHTML=`<div class="wrap"><div class="sec-title">＋ 新开场次</div>
    <div class="field"><label>剧本</label><select id="f-script">${scripts.map(s=>`<option value="${s.id}" data-p="${s.player_max}">${esc(s.title)}</option>`).join('')}</select></div>
    <div class="field"><label>DM</label><select id="f-dm"><option value="">未派单</option>${dms.map(d=>`<option value="${d.id}">${esc(d.nickname)}</option>`).join('')}</select></div>
    <div class="field"><label>开场时间</label><input id="f-time" type="datetime-local"></div>
    <div class="field"><label>人数</label><input id="f-need" type="number" value="6"></div>
    <div class="field"><label>价格</label><input id="f-price" type="number" value="198"></div>
    <button class="btn block" onclick="submitNew()">发布场次</button>
    <button class="btn block ghost" style="margin-top:8px" onclick="renderSchedule()">取消</button></div>`;
}
async function submitNew(){
  const t=document.getElementById('f-time').value;
  if(!t){alert('请选择开场时间');return;}
  const time=t.replace('T',' ');
  const r=await API.post('/api/merchant/schedule',{
    shop_id:SHOP_ID, script_id:+document.getElementById('f-script').value,
    dm_id:document.getElementById('f-dm').value?+document.getElementById('f-dm').value:null,
    start_time:time, need_players:+document.getElementById('f-need').value,
    price:+document.getElementById('f-price').value});
  if(r.error)alert(r.error);else{alert('场次已发布，玩家可在拼车广场看到并上车');renderSchedule();}
}

async function renderScripts(){
  app.innerHTML=`<div class="wrap"><div class="sec-title">📚 剧本库</div><div id="sl" class="loading">加载中…</div></div>`;
  const list=await API.get('/api/player/scripts');
  document.getElementById('sl').innerHTML=list.map(s=>`<div class="card script-card">
    <img class="cover" src="${s.cover}" onerror="this.style.opacity=.3">
    <div class="info"><h3>${esc(s.title)} <span class="rate" style="font-size:13px">★${s.rating}</span></h3>
    <div class="meta">${s.category} · ${s.player_min}-${s.player_max}人 · 玩过${s.play_count}次</div>
    <div>${(s.tags||[]).map(t=>`<span class="tag">${esc(t)}</span>`).join('')}</div></div></div>`).join('');
}

async function renderDMs(){
  app.innerHTML=`<div class="wrap"><div class="sec-title">🎙️ DM 花名册</div><div id="dl" class="loading">加载中…</div></div>`;
  const list=await API.get(`/api/merchant/shops/${SHOP_ID}/dms`);
  document.getElementById('dl').innerHTML=list.map(d=>`<div class="card">
    <div class="row" style="align-items:center"><div style="width:44px;height:44px;border-radius:50%;background:var(--card2);display:flex;align-items:center;justify-content:center;font-size:20px">🎙️</div>
    <div style="flex:1"><h3 style="font-size:15px">${esc(d.nickname)} <span class="rate" style="font-size:12px">★${d.rating}</span></h3>
    <div class="meta">擅长：${esc(d.good_at)} · ${d.fans}粉丝</div></div>
    <span class="status ${d.available?'recruiting':'full'}">${d.available?'可接单':'休息中'}</span></div></div>`).join('');
}

initAiBadge(); go('dash');

File: cloud-functions/backend/routes/merchant.py
"""店家端 API：排期管理、剧本库、DM 派单、订单概览、数据看板。"""
from datetime import datetime
from flask import Blueprint, request, jsonify
from ..models import (db, User, Shop, Script, DMProfile, GameSession, SessionMember)

merchant_bp = Blueprint("merchant", __name__, url_prefix="/api/merchant")


@merchant_bp.route("/shops/<int:owner_id>")
def my_shops(owner_id):
    items = Shop.query.filter_by(owner_id=owner_id).all()
    return jsonify([s.to_dict() for s in items])


# ---------------- 排期管理 ----------------
@merchant_bp.route("/shops/<int:shop_id>/schedule")
def shop_schedule(shop_id):
    """门店所有场次（排期看板）。"""
    items = GameSession.query.filter_by(shop_id=shop_id).order_by(GameSession.start_time).all()
    out = []
    for s in items:
        d = s.to_dict()
        script = Script.query.get(s.script_id)
        dm = DMProfile.query.get(s.dm_id) if s.dm_id else None
        d["script_title"] = script.title if script else ""
        d["dm_name"] = (User.query.get(dm.user_id).nickname if dm else "未派单")
        out.append(d)
    return jsonify(out)


@merchant_bp.route("/schedule", methods=["POST"])
def create_schedule():
    """店家开一个场次（可供玩家上车）。"""
    data = request.json or {}
    try:
        start = datetime.strptime(data["start_time"], "%Y-%m-%d %H:%M")
    except Exception:
        return jsonify({"error": "时间格式错误，应为 YYYY-MM-DD HH:MM"}), 400
    s = GameSession(
        script_id=data["script_id"], shop_id=data["shop_id"],
        dm_id=data.get("dm_id"), host_id=data.get("host_id", 1),
        start_time=start, price=data.get("price", 0),
        need_players=data.get("need_players", 6), joined_players=0,
        status="recruiting", note=data.get("note", "店家开场"),
    )
    db.session.add(s)
    db.session.commit()
    return jsonify(s.to_dict())


@merchant_bp.route("/schedule/<int:sid>/status", methods=["POST"])
def update_status(sid):
    s = GameSession.query.get_or_404(sid)
    s.status = (request.json or {}).get("status", s.status)
    db.session.commit()
    return jsonify(s.to_dict())


@merchant_bp.route("/schedule/<int:sid>/assign_dm", methods=["POST"])
def assign_dm(sid):
    s = GameSession.query.get_or_404(sid)
    s.dm_id = (request.json or {}).get("dm_id")
    db.session.commit()
    return jsonify(s.to_dict())


# ---------------- 剧本库 ----------------
@merchant_bp.route("/scripts", methods=["POST"])
def add_script():
    data = request.json or {}
    s = Script(
        title=data["title"], category=data.get("category", ""),
        tags=",".join(data.get("tags", [])) if isinstance(data.get("tags"), list) else data.get("tags", ""),
        player_min=data.get("player_min", 6), player_max=data.get("player_max", 6),
        duration=data.get("duration", 240), difficulty=data.get("difficulty", 3),
        is_be=data.get("is_be", False), has_horror=data.get("has_horror", False),
        intro=data.get("intro", ""), cover=data.get("cover", ""),
    )
    db.session.add(s)
    db.session.commit()
    return jsonify(s.to_dict())


# ---------------- DM 花名册 ----------------
@merchant_bp.route("/shops/<int:shop_id>/dms")
def shop_dms(shop_id):
    items = DMProfile.query.filter_by(shop_id=shop_id).all()
    out = []
    for x in items:
        d = x.to_dict()
        u = User.query.get(x.user_id)
        d["nickname"] = u.nickname if u else "DM"
        out.append(d)
    return jsonify(out)


# ---------------- 数据看板 ----------------
@merchant_bp.route("/shops/<int:shop_id>/dashboard")
def dashboard(shop_id):
    sessions = GameSession.query.filter_by(shop_id=shop_id).all()
    total = len(sessions)
    confirmed = len([s for s in sessions if s.status in ("full", "confirmed", "finished")])
    total_players = sum(s.joined_players for s in sessions)
    revenue = sum(s.price * s.joined_players for s in sessions if s.status in ("full", "confirmed", "finished"))
    # 热门本
    from collections import Counter
    c = Counter(s.script_id for s in sessions)
    hot = []
    for sid, cnt in c.most_common(3):
        sc = Script.query.get(sid)
        if sc:
            hot.append({"title": sc.title, "count": cnt})
    return jsonify({
        "total_sessions": total,
        "confirmed_sessions": confirmed,
        "total_players": total_players,
        "revenue": round(revenue, 0),
        "fill_rate": round(confirmed / total * 100, 0) if total else 0,
        "hot_scripts": hot,
    })
